src/main/java/picard/vcf/GenotypeConcordance.java

```diff
@@ -457,70 +584,89 @@ public static void outputDetailMetricsFile(final VariantContext.Type variantType
+     * no first or second allele was present respectively. */
+    private static class Alleles {
+        public final OrderedSet<String> allAlleles;
+        public final String truthAllele1;
+        public final String truthAllele2;
+        public final String callAllele1;
+        public final String callAllele2;
+
+
+        public Alleles(final OrderedSet<String> allAlleles,
+                       final String truthAllele1,
+                       final String truthAllele2,
+                       final String callAllele1,
+                       final String callAllele2) {
+            this.allAlleles   = allAlleles;
+            this.truthAllele1 = (truthAllele1 == null) ? null : this.allAlleles.get(allAlleles.indexOf(truthAllele1));
+            this.truthAllele2 = (truthAllele2 == null) ? null : this.allAlleles.get(allAlleles.indexOf(truthAllele2));
+            this.callAllele1  = (callAllele1 == null) ? null : this.allAlleles.get(allAlleles.indexOf(callAllele1));
+            this.callAllele2  = (callAllele2 == null) ? null : this.allAlleles.get(allAlleles.indexOf(callAllele2));
+        }
 
-        // TODO: what about getPloidy()
 
-        Genotype truthGenotype = null, callGenotype = null;
+        public List<Allele> asAlleles() {
+            if (allAlleles.isEmpty()) {
+                return Collections.emptyList();
+            }
+            else {
+                final List<Allele> alleles = new ArrayList<>(4);
+                int idx = 0;
+                while (idx < this.allAlleles.size()) {
+                    alleles.add(Allele.create(this.allAlleles.get(idx), idx == 0));
+                    idx += 1;
+                }
+                return alleles;
+            }
+        }
 
-        // Site level checks
-        // Added potential to include missing sites as hom ref..
-        if (truthContext == null) truthState = TruthState.MISSING;
-        else if (truthContext.isMixed()) truthState = TruthState.IS_MIXED;
-        else if (truthContext.isFiltered()) truthState = TruthState.VC_FILTERED;
-        else {
-            // Genotype level checks
-            truthGenotype = truthContext.getGenotype(truthSample);
-            if (truthGenotype.isNoCall())           truthState = TruthState.NO_CALL;
-            else if (truthGenotype.isFiltered())    truthState = TruthState.GT_FILTERED;
-            else if ((truthGenotype.getGQ() != -1) && (truthGenotype.getGQ() < minGq)) truthState = TruthState.LOW_GQ;
-            else if ((truthGenotype.getDP() != -1) && (truthGenotype.getDP() < minDp)) truthState = TruthState.LOW_DP;
-            // Note.  Genotype.isMixed means that it is called on one chromosome and NOT on the other
-            else if ((truthGenotype.isMixed())) truthState = TruthState.NO_CALL;
+        public List<Allele> truthAlleles() {
+            if (allAlleles.isEmpty() || this.truthAllele1 == null) {
```

Comment:
It feels a little odd that in the constructor you check `truthAllele2` for null independently, but here you seem to assume it must be non-null if `truthAllele1` is.